apollo-federation/src/sources/connect/json_selection/README.md

```diff
@@ -99,7 +99,8 @@ PathStep             ::= "." Key | "->" Identifier MethodArgs?
 Key                  ::= Identifier | LitString
 Identifier           ::= [a-zA-Z_] NO_SPACE [0-9a-zA-Z_]*
 MethodArgs           ::= "(" (LitExpr ("," LitExpr)* ","?)? ")"
-LitExpr              ::= LitPrimitive | LitObject | LitArray | PathSelection
+LitExpr              ::= LitPath | LitPrimitive | LitObject | LitArray | PathSelection
+LitPath              ::= (LitPrimitive | LitObject | LitArray) PathStep+
```

Comment:
It's important that `LitPath` is tried first in the list of `LitExpr` alternatives, because a `LitPath` always has a prefix that looks like one of the other JSON value types.

Reply:
It's also important that we never allow a `PathSelection` to appear at the beginning of the `LitPath` (only `(LitPrimitive | LitObject | LitArray)`), because then the boundary between the `PathSelection` and the `PathStep+` suffix would be ambiguous.

Reply:
is this a comment worth leaving in the readme? or is it sufficiently covered in code comments?